src/analysis/retail/warrior/arms/normalizers/ImprovedExecuteNormalizer.ts

```diff
@@ -0,0 +1,85 @@
+import getRage from 'analysis/retail/warrior/shared/utils/getRage';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_WARRIOR } from 'common/TALENTS';
+import MAGIC_SCHOOLS from 'game/MAGIC_SCHOOLS';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { AnyEvent, EventType, ResourceActor, ResourceChangeEvent } from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import { damageEvent as executeDamageEvent } from './ExecuteLinkNormalizer';
+
+/**
+ * Normalizer that fabricates ResrouceChange events for the Improved Execute talent.
+ *
+ * Whenever Execute lands without killing the target, return 10% of the rage cost.
+ *
+ * Attributes the Rage refund to the Improved Execute talent.
+ */
+export default class ImprovedExecuteNormalizer extends EventsNormalizer {
+  normalize(events: AnyEvent[]): AnyEvent[] {
+    const updatedEvents: AnyEvent[] = [];
+
+    for (const event of events) {
+      updatedEvents.push(event);
+
+      if (
+        event.type === EventType.Cast &&
+        event.sourceID === this.selectedCombatant.id &&
+        (event.ability.guid === SPELLS.EXECUTE.id ||
+          event.ability.guid === SPELLS.EXECUTE_GLYPHED.id)
+      ) {
+        const rage = getRage(event, this.selectedCombatant);
+
+        if (rage?.cost != null && rage.cost > 0) {
+          const damageEvent = executeDamageEvent(event);
+          if (damageEvent?.overkill == null || damageEvent?.overkill < 0) {
+            // Possible this should be .floor() instead of .round()
+            const refund = Math.round(rage.cost * 0.1);
```

Comment:
round is probably close enough absent any direct proof one way or the other.

Reply:
I kept having rounding mismatchews every now and then however I rounded it :'( 

Actual rage is probably a float, but they log it as rounded values. I thought the whole 100 rage = 1000 rage was to not have decimals :( But yeah rounded is probably as good as it gets. Could remove this comment

Reply:
yea, sounds similar to other resources then. we just do the best we can with the logged info